x-pack/platform/plugins/private/reporting/server/lib/tasks/run_single_report.ts

```diff
@@ -38,37 +33,14 @@ export class RunSingleReportTask extends RunReportTask<ReportTaskParams> {
     }
 
     const m = moment();
-
```

Comment:
Previously, the single report task retries report generation 3 times but it actually ran the task 4 times. If the first 3 runs fail, the 4th run just updates the report document with a `failed` status.

This PR updates the logic to update the report document with a `failed` status at the end of the 3rd run (the last attempt) if it fails. This allows the scheduled reporting task to use the same logic to update the report document with a `failed` status at the end of its run.